### PR TITLE
Easels and mirrors are craftable; mirrors are also properly deconstructable.

### PR DESCRIPTION
**Pull request by harmonyn**
:cl: bandit
add: Easels and mirrors are craftable/buildable. Mirrors can be properly deconstructed.
/:cl:

Slowly but surely, more of the station becomes replicable.

### CODE REVIEW

File: code/game/objects/structures/mirror.dm [hunk 1]
```diff
@@ -92,6 +95,47 @@
...
+	add_fingerprint(user)
+	if(istype(W, /obj/item/stack/sheet/glass))
+		if(!W.tool_start_check(user, amount=2))
+			return
+		playsound(src, 'sound/items/deconstruct.ogg', 50, 1)
+		to_chat(user, "<span class='notice'>You start to add glass to the mirror frame...</span>")
+		if(W.use_tool(src, user, 20, amount=2))
+			to_chat(user, "<span class='notice'>You put in the glass panel.</span>")
+			newmirror = new /obj/structure/mirror(loc)
+			if(pixel_x)
```
Comment (4dplanner) [inline code/game/objects/structures/mirror.dm]: Pretty sure this if isn't actually needed

Comment (bgobandit) [reply code/game/objects/structures/mirror.dm]: on attach() it can be x or y dependent on the dir

Comment (4dplanner) [reply code/game/objects/structures/mirror.dm]: Yes, but setting both of them has the same effect - consider the case where pixel_x = 0, so we go to the else branch. not setting pixel_x has the exact same effect as setting it to 0, so it makes no difference.

Comment (bgobandit) [reply code/game/objects/structures/mirror.dm]: but if the pixel shift is on pixel_y something has to account for that

**Comment by anconfuzedrock** [comment]
Soom poor poor baby will create an easel with his finest wood, learn easels are useless cuz painting is removed, and cry for 75 years.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by PKPenguin321** [comment]
>can't nuke snowflakes by shaving their head and smashing all the mirrors anymore

Sad

**Comment by bgobandit** [comment]
@cyberboss idk why it says changes requested still but I did them

**Comment by coiax** [comment]
Tbh, my headcanon was always that mirrors required silver to construct, to make them less easy to construct.

But I worry if they're deconstructable, people will just grab all the mirrors for that silver.

**Comment by bgobandit** [comment]
mirrors don't really have a huge in-game benefit so I don't know why they'd need to be gated behind silver

I considered r-glass though since breaking them is pretty difficult

File: code/game/objects/structures/mirror.dm [hunk 2]
```diff
@@ -60,11 +63,22 @@
...
-		if(!disassembled)
-			new /obj/item/shard( src.loc )
+/obj/structure/mirror/crowbar_act(mob/user, obj/item/I)
+	playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+	user.visible_message("[user] begins to remove the glass from [src].", "<span class='notice'>You begin to take the glass off [src]...</span>")
+	if(I.use_tool(src, user, 50))
+		playsound(src.loc, 'sound/items/deconstruct.ogg', 50, 1)
+		user.visible_message("[user] removes the glass from [src].", "<span class='notice'>You remove the glass from [src].</span>")
+	if (broken)
+		new /obj/item/shard(user.loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/mirror.dm]:
```suggestion
		new /obj/item/shard(user.drop_location())
```

File: code/game/objects/structures/mirror.dm [hunk 3]
```diff
@@ -60,11 +63,22 @@
...
+	else
+		new /obj/item/stack/sheet/glass(user.loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/mirror.dm]:
```suggestion
		new /obj/item/stack/sheet/glass(user.drop_location())
```

File: code/game/objects/structures/mirror.dm [hunk 4]
```diff
@@ -60,11 +63,22 @@
...
 
-/obj/structure/mirror/deconstruct(disassembled = TRUE)
-	if(!(flags_1 & NODECONSTRUCT_1))
-		if(!disassembled)
-			new /obj/item/shard( src.loc )
+/obj/structure/mirror/crowbar_act(mob/user, obj/item/I)
+	playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+	user.visible_message("[user] begins to remove the glass from [src].", "<span class='notice'>You begin to take the glass off [src]...</span>")
+	if(I.use_tool(src, user, 50))
+		playsound(src.loc, 'sound/items/deconstruct.ogg', 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/mirror.dm]:
```suggestion
		playsound(src, 'sound/items/deconstruct.ogg', 50, 1)
```

File: code/game/objects/structures/mirror.dm [hunk 5]
```diff
@@ -60,11 +63,22 @@
 			desc = "Oh no, seven years of bad luck!"
 		broken = TRUE
 
-/obj/structure/mirror/deconstruct(disassembled = TRUE)
-	if(!(flags_1 & NODECONSTRUCT_1))
-		if(!disassembled)
-			new /obj/item/shard( src.loc )
+/obj/structure/mirror/crowbar_act(mob/user, obj/item/I)
+	playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/mirror.dm]:
```suggestion
	playsound(src, 'sound/machines/click.ogg', 50, 1)
```

File: code/game/objects/structures/mirror.dm [hunk 6]
```diff
@@ -60,11 +63,22 @@
 			desc = "Oh no, seven years of bad luck!"
 		broken = TRUE
 
-/obj/structure/mirror/deconstruct(disassembled = TRUE)
-	if(!(flags_1 & NODECONSTRUCT_1))
-		if(!disassembled)
-			new /obj/item/shard( src.loc )
+/obj/structure/mirror/crowbar_act(mob/user, obj/item/I)
```
Comment (Cyberboss) [inline code/game/objects/structures/mirror.dm]:
```suggestion
/obj/structure/mirror/crowbar_act(mob/living/user, obj/item/I)
```

File: code/game/objects/structures/mirror.dm [hunk 7]
```diff
@@ -92,6 +106,60 @@
...
+		var/obj/item/newframe = new /obj/item/wallframe/mirror(drop_location())
+		transfer_fingerprints_to(newframe)
+		qdel(src)
+		return
+
+
+/obj/structure/mirror_construct/deconstruct(disassembled = TRUE)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		if(!disassembled)
+			new /obj/item/stack/sheet/metal(loc, 3)
```
Comment (Cyberboss) [inline code/game/objects/structures/mirror.dm]:
```suggestion
			new /obj/item/stack/sheet/metal(drop_location(), 3)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by bgobandit** [comment]
I will try to get to this tomorrow

**Comment by bgobandit** [comment]
ok changed the shit

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
Your PR is getting old. Either Travis CI is failing, it is still labelled work in progress (change this by adding `[READY]` to the start of your title), or there are unaddressed change requests. This may be closed soon to free up space on our tracker if it remains idle. In the event that it is, please ask in #coderbus on Rizon for it to be reopened once you have resumed work on it.

**Pull request closed without merging**